### PR TITLE
remove misleading "ERROR" from AcceptToMemoryPool debug.log lines

### PR DESCRIPTION
**Pull request by LarryRuane**
Addresses #2457, remove "ERROR" from debug.log "AcceptToMemoryPool: inputs already spent"

### CODE REVIEW

**Comment by LarryRuane** [comment]
#2457 lists two examples of normal-operation messages that contain `ERROR`; this PR fixes the first one; the second one (" JoinSplit is dependent on an anchor that gets orphaned") is no longer applicable.

These two are *examples* -- I don't know what other instances of this problem may exist, nor do I know how to find them (without having a deep understanding of the system). That's why this PR only *addresses* the issue.

**Comment by LarryRuane** [comment]
Force-pushed a fix for another instance of this problem, example:
```
2018-08-21 04:03:17 ERROR: AcceptToMemoryPool: joinsplit requirements not met
```

**Comment by LarryRuane** [comment]
Is it worth having a unit test for this? It would be easy to do but I'm not sure it's worth it.

**Comment by LarryRuane** [comment]
Rebased with master, the RPC test that this PR includes still passes (didn't run the full test suite, but highly unlikely that it wouldn't pass).

File: qa/rpc-tests/already_spent.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python2
+# Copyright (c) 2018 The Zcash developers
```
Comment (rex4539) [inline qa/rpc-tests/already_spent.py]: 2019 perhaps? :)

**Comment by LarryRuane** [comment]
Rebased, can squash the copyright commit

**Comment by LarryRuane** [comment]
Rebased from latest master, fixed merge conflicts.

**Comment by daira** [comment]
This needs to be rebased before reviewing, since it touches consensus code.

**Comment by daira** [comment]
Ping

**Comment by str4d** [comment]
This is the wrong fix to the problem. Rather than logging all consensus failures at the `INFO` level instead of the `ERROR` level, we should be selecting the level based on the conditions:
- If `IsInitialBlockDownload`, we should probably log at `DEBUG` (or just drop those transactions/blocks on the floor early, so we never reach here).
- If we are calling from `AcceptToMemoryPool`, `INFO` seems reasonable.
- If we are calling from `ProcessNewBlock`, we should use `WARNING` instead of `ERROR` (as while these are consensus violations, they are not hard errors for the node).

This would be a reasonably large refactor, so we should also take the time to leverage the newer `tracing` features for generally better logging.

**Pull request closed without merging**